### PR TITLE
FIX: Was unable to import key on macOS

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/86zgpwfaqfajnzxx2402x3eham

[download apk](http://filestorage.bluewallet.io:1488/1607550716.apk)

File: screen/send/psbtMultisigQRCode.js [hunk 1]
```diff
@@ -7,9 +7,13 @@ import { SquareButton } from '../../components/SquareButton';
 import { getSystemName } from 'react-native-device-info';
 import loc from '../../loc';
 import ImagePicker from 'react-native-image-picker';
-import ScanQRCode from './ScanQRCode';
 import { useNavigation, useRoute, useTheme } from '@react-navigation/native';
+import RNFS from 'react-native-fs';
```
Comment (Overtorment) [inline screen/send/psbtMultisigQRCode.js:11]: can we hide all that RNFS and document picker biz inside `../../blue_modules/fs` ?

File: screen/send/psbtMultisigQRCode.js [hunk 2]
```diff
@@ -41,35 +45,97 @@ const PsbtMultisigQRCode = () => {
...
+    if (Platform.OS === 'ios') {
+      const options = [loc._.cancel, 'Take Photo', loc.wallets.list_long_choose, loc.wallets.import_file];
+
+      ActionSheet.showActionSheetWithOptions({ options, cancelButtonIndex: 0 }, buttonIndex => {
+        if (buttonIndex === 1) {
+          takePhoto();
+        } else if (buttonIndex === 2) {
+          choosePhoto();
+        } else if (buttonIndex === 3) {
+          handleImportFileButtonPressed();
```
Comment (Overtorment) [inline screen/send/psbtMultisigQRCode.js:130]: better naming for those 3 methods would be helpful, e.g. `takePhotoMacos` & `choosePhotoMacos`

File: screen/send/psbtMultisigQRCode.js [hunk 3]
```diff
@@ -41,35 +45,97 @@ const PsbtMultisigQRCode = () => {
...
+        }
+      });
+    }
```
Comment (Overtorment) [inline screen/send/psbtMultisigQRCode.js:133]: `else` for `(Platform.OS === 'ios')` not handled? can it be an android tablet?

File: screen/send/psbtMultisigQRCode.js [hunk 4]
```diff
@@ -41,35 +45,97 @@ const PsbtMultisigQRCode = () => {
...
       // we dont support it in this flow
+      alert(loc.wallets.import_error);
     } else {
       // psbt base64?
       navigate('PsbtMultisig', { receivedPSBTBase64: ret.data });
     }
   };
 
+  const choosePhoto = () => {
+    ImagePicker.launchImageLibrary(
```
Comment (Overtorment) [inline screen/send/psbtMultisigQRCode.js:56]: maybe imagepicker biz should also be abstracted and hidden in a dedicated module, e.g.  `blue_modules/imagepicker`
or maybe in existing `blue_modules/fs`

File: screen/wallets/addMultisigStep2.js [hunk 5]
```diff
@@ -427,6 +409,87 @@ const WalletsAddMultisigStep2 = () => {
...
+      } else {
+        throw new Error();
+      }
+    } catch (err) {
+      if (!DocumentPicker.isCancel(err)) {
+        alert(loc.wallets.import_error);
+      }
+    }
+  };
+
```
Comment (Overtorment) [inline screen/wallets/addMultisigStep2.js:492]: same actionsheet, same biz with documentpicker, iamgepigcker etc.
DRY.
see my prev comment about blue_modules

**Review by Overtorment — CHANGES_REQUESTED** [review]

**Comment by marcosrdz** [comment]
@Overtorment  moved code to fs.

File: blue_modules/fs.js [hunk 6]
```diff
@@ -141,3 +193,5 @@ const showFilePickerAndReadFile = async function () {
 module.exports.writeFileAndExport = writeFileAndExport;
 module.exports.openSignedTransaction = openSignedTransaction;
 module.exports.showFilePickerAndReadFile = showFilePickerAndReadFile;
+module.exports.showImagePickerAndReadImage = showImagePickerAndReadImage;
+module.exports.takePhotoWithImagePickerAndReadPhoto = takePhotoWithImagePickerAndReadPhoto;
```
Comment (Overtorment) [inline blue_modules/fs.js:197]: noice. clean

File: screen/wallets/addMultisigStep2.js [hunk 7]
```diff
@@ -426,6 +403,23 @@ const WalletsAddMultisigStep2 = () => {
...
+        fs.showImagePickerAndReadImage(onBarScanned).catch(error => alert(error.message));
+      } else if (buttonIndex === 3) {
+        const { data } = await fs.showFilePickerAndReadFile();
+        if (data) {
+          onBarScanned({ data });
+        }
+      }
+    });
+  };
+
```
Comment (Overtorment) [inline screen/wallets/addMultisigStep2.js:422]: `showActionSheet` is duplicated in 2 places.  at least its simple enough to read
doest it make sense to move it to `fs`..?
you decide on this. for now its good enough.

**Review by Overtorment — APPROVED** [review]

File: screen/wallets/addMultisigStep2.js (hunk 7 above)
Comment (marcosrdz) [reply screen/wallets/addMultisigStep2.js:422]: Different screens have different action sheet items. its the only reason i kept it this way.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/cv4jgvwp646kzp184ydt5abxw8

[download apk](http://filestorage.bluewallet.io:1488/1607798595.apk)

**Review by ncoelho — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**